src/Engine/ProtoCore/DSASM/Executive.cs

```diff
@@ -5409,7 +5409,17 @@ private void MOD_Handler(Instruction instruction)
             {
                 if (opdata1.IsInteger && opdata2.IsInteger)
                 {
-                    opdata2 = StackValue.BuildInt(opdata2.RawIntValue % opdata1.RawIntValue);
+                    long lhs = opdata2.RawIntValue;
+                    long rhs = opdata1.RawIntValue;
+                    if (rhs == 0)
```

Comment:
I havent really tested other languages but is it only rhs that need to be handled by 0?